File: problems/two-pointers/src/find_the_index_of_the_first_occurrence_in_a_string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        n = len(haystack)
        m = len(needle)
        for l in range(n - m + 1):
            r = 0
            while r < m and haystack[l + r] == needle[r]:
                r += 1
            if r == m:
                return l
        return -1

File: problems/two-pointers/solutions/java/FindTheIndexOfTheFirstOccurrenceInAString.java
package java;

public class FindTheIndexOfTheFirstOccurrenceInAString {
    public int strStr(String haystack, String needle) {
        int n = haystack.length();
        int m = needle.length();
        for (int i = 0; i < n - m + 1; i++) {
            int l = i;
            int r = 0;
            while (r < m) {
                if (haystack.toCharArray()[l] != needle.toCharArray()[r]) {
                    break;
                }
                l++;
                r++;
            }
            if (r == m) {
                return i;
            }
        }
        return -1;
    }
}
